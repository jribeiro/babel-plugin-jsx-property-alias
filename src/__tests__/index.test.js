// @flow
import { readdirSync } from 'fs';
import { join } from 'path';
import { transformFileSync } from 'babel-core';

import getPluginOptionsForDirectory from './getPluginOptionsForDirectory';
import plugin from '../';

describe('fixtures', () => {
  const fixturesDir = join(__dirname, 'fixtures');
  readdirSync(fixturesDir).forEach((caseName) => {
    it(`should work with ${caseName.split('-').join(' ')}`, () => {
      const fixtureDir = join(fixturesDir, caseName);
      const pluginOptions = getPluginOptionsForDirectory(fixtureDir);

      // Only run plugins targeting the production env.
      // process.env.BABEL_ENV = 'production';
      const actual = transformFileSync(join(fixtureDir, 'actual.js'), {
        babelrc: false,
        plugins: [
          [plugin, pluginOptions],
          '@babel/plugin-syntax-jsx',
        ],
      }).code;

      expect(actual).toMatchSnapshot();
    });
  });
});
