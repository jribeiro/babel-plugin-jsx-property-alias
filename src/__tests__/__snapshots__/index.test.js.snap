// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures should work with alias 1`] = `
"React.createClass({
  render: function () {
    return <div className=\\"bar\\" testID=\\"appiumSelector\\" accessibilityLabel=\\"appiumSelector\\">
        Hello Wold!
      </div>;
  }
});"
`;

exports[`fixtures should work with class 1`] = `
"class MyComponent extends React.Component {
  render() {
    return <div testID=\\"foo\\" accessibilityLabel=\\"foo\\" accessibilityContent=\\"foo\\" />;
  }

}"
`;

exports[`fixtures should work with complex 1`] = `
"React.createClass({
  render: function ({
    testID
  }) {
    return <button testID={testID} foo=\\"foo\\" accessibilityLabel={testID} accessibilityContent={testID} bar=\\"foo\\" />;
  }
});
React.createClass({
  render: function () {
    return <Foo testID=\\"bar\\" accessibilityLabel=\\"bar\\" accessibilityContent=\\"bar\\" />;
  }
});"
`;

exports[`fixtures should work with noop 1`] = `
"React.createClass({
  render: function () {
    return <div className=\\"bar\\" testID=\\"appiumSelector\\">
        Hello Wold!
      </div>;
  }
});"
`;

exports[`fixtures should work with spread 1`] = `
"const Button = React.createClass({
  render: function ({
    testID
  }) {
    return <button testID={testID} accessibilityLabel={testID} />;
  }
});
const Main = React.createClass({
  render: function (props) {
    return <Button {...props} />;
  }
});
React.createElement(Main, {
  testID: 'foo'
});"
`;
